### ISSUE TITLE
Inconsistent Accessibility Behavior Across Platforms in MAUI

### ISSUE THREAD

**Issue by KamalaKannan-SF4101** [dotnet/maui#33612]
### Description

We are using a StackLayout with a BindableLayout collection. Inside the item template, each item contains a Border with a Label, and the Border has gesture recognizers attached. We are experiencing inconsistent accessibility behavior across platforms:
 
**iOS and macOS:**
The accessibility focus does not move to each individual item in the collection when the parent element has SemanticProperties applied. Even we try to  set Semantics to the ContentView , Scrollview and  Stacklayout,
 
**Windows:**
Accessibility focus reaches each item, but the interaction (tap/activate) does not work as expected. We have use [CapsLock + Enter] and [CapsLock+space] to interact but it does'nt even work.

**Android**
For the border when both  Semantics "Description and Hint" is set , it reads properly but, due to the  border gesture again it reads "Double tap to activate" 
 
 Please investigate and provide a solution to ensure consistent accessibility behavior across all supported platforms.

### Steps to Reproduce

1. Run the attached sample on:
   iOS
   macOS
   Windows
   Android
2. Enable the respective screen reader:
     VoiceOver on iOS/macOS    
     Narrator on Windows
3.  Attempt to navigate and interact with each item in the BindableLayout

[MauiSample.zip](https://github.com/user-attachments/files/24728040/MauiSample.zip)

### Version with bug

10.0.30

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows, macOS

### Relevant log output

```shell

```

**Comment by HarishKumarSF4517** [comment]
This issue has been validated in Visual Studio Code 1.108.0 and Visual Studio 18.1.1 with MAUI versions 9.0.120, 10.0.20 and 10.0.30  It was reproducible on Windows, Android, iOS and macOS platform.

**Comment by KillswitchPrime** [comment]
I have encountered the same issue. Any element that has children will be read by VoiceOver (iOS) as "Description for element not available", but will read out the children ok. If you add SemanticProperties Hint or Description on the parent, VoiceOver will read this out loud, but it is no longer possible to focus on any of the child elements. This creates some very user unfriendly experiences.
On Android (TalkBack) this behavior seems to be mostly ok with each parent being given a short description (CarouselView gets the hint "swipe left or right" automatically for example), and then reading out the child elements.
